src/main/scala/io/iohk/ethereum/network/p2p/messages/Capability.scala

```diff
@@ -7,6 +7,21 @@ import io.iohk.ethereum.rlp.{RLPEncodeable, RLPException, RLPList, RLPSerializab
 case class Capability(name: String, version: Byte)
 
 object Capability {
+  def negotiate(c1: List[Capability], c2: List[Capability]): Option[Capability] =
+    c1.intersect(c2).maxByOption(_.version) // FIXME ignores branches and other protocols
```

Comment:
It seems that it duplicates the `best` function here.
We should probably have a single point where the best protocol is chosen, no ?
Something like :
```suggestion
   // FIXME ignores branches and other protocols
   c1.intersect(c2) match {
     case Nil => None
     case nonEmpty => best(nonEmpty)
   }
```